src/Shared/runtime/Http2/Hpack/HPackDecoder.cs

```diff
@@ -430,6 +430,11 @@ private void ParseHeaderName(ReadOnlySpan<byte> data, ref int currentIndex, IHtt
                 currentIndex += count;
 
                 _state = State.HeaderValueLength;
+                ParseHeaderValueLength(data, ref currentIndex, handler);
+            }
+            else if (count == 0)
+            {
+                // no-op
```

Comment:
It is extremely uncommon. It will only happen if headers are read in chunks and the header value length is at the end of a chunk and the value is in the next chunk.

Think it should be removed?

Reply:
I don't think it needs to be removed. I was just curious if we could expect an improvement.